src/main.py

```diff
@@ -3,60 +3,110 @@
-    # Create a DataFrame
-    data = pd.DataFrame({
-        'Patient ID': range(total_patients),
-        'Propensity Score': propensity_scores,
-        'Treatment': treatment_assignment
-    })
-    return data
-
 def main():
     logger = logging.getLogger('Main')
     logger.debug('======Start======')
+    
     # Generate synthetic medical dataset with additional columns and binary outcome
     # Generate the dataset
-    total_patients = 100
-    treated_patients = 20
-    medical_data = generate_dataset(total_patients, treated_patients)
-    matched_pairs = methods.nnm2(medical_data, replacement=True, caliper=0.02, k_neighbors=1, method='caliper')
+    """
+    :param treated_count: amount of treated patients
+    :param untreated_count: amount of untreated patients
+    :param num_params: number of numerical parameters (ie: age)
+    :param cat_params: number of categorical parameters (ie: race)
+    :param num_categories: number of categories to base the categorical parameters on
+    :return: pd.DataFrame: returns the data frame created by the method
+    """
+    treated_patients = 200
+    untreated_count = 800
+    
+    num_params = 50
+    cat_params = 50
+    num_categories = 5
     
-    folder_name = "build"
-    file_name = "matched_pairs.txt"
+    matched_pairs = pd.DataFrame()
+    result_df = DataGenerator.generate_data(treated_patients, untreated_count, num_params, cat_params, num_categories)
+    print(result_df) 
+    samples = [(1, 0), (0, 1), (1, 1), (5, 0), (0, 5), (5, 5), (50, 0), (0, 50), (50, 50)]
+
+ 
+    target = [dd.treatment]
+    folder_name = "build/"
     if not os.path.exists(folder_name):
         os.mkdir(folder_name)
+    for treatment in samples:
+        #build the column labels to be passed to logistic regression for testing purposes
+        combined_column_names = []
+        x, y = treatment 
+        if x >  0:
+            selected_numerical_column_names = result_df.columns[1:x+1]
+            combined_column_names += selected_numerical_column_names.tolist()
+        if y > 0:
+            selected_categorical_column_names = result_df.columns[num_params+1:y+51]
+            combined_column_names += selected_categorical_column_names.tolist()
+        # Combine both sets of column names
+        #create a file for each sample combination
+        timestamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
+        file_name = f'match_numerical_{x}_categorical_{y}_{timestamp}.csv'
+        file_path = os.path.join(folder_name, file_name)
+        #calculate psm scores and return a new dataframe of just the sample columns with patient id and psm scores
+        data = LogReg.LogRegress(result_df, combined_column_names, target)
+        #calculate the pairs and save them to file
+        matched_pairs = methods.nnm2(data, replacement=True, caliper=0.02, k_neighbors=1, method='caliper')
+        
+        logger.debug(f'pairs: \n{matched_pairs}')
```

Comment:
Do not log all matched pairs. Save it in the file. If you have 2K matched records this log is not useful.